packages/contracts/contracts/gatekeepers/SignUpGatekeeper.sol

```diff
@@ -1,20 +1,16 @@
 // SPDX-License-Identifier: MIT
 pragma solidity ^0.8.28;
 
+import { Policy } from "@excubiae/contracts/policy/Policy.sol";
+import { IBasePolicy } from "@excubiae/contracts/interfaces/IBasePolicy.sol";
+
 /// @title SignUpGatekeeper
 /// @notice A gatekeeper contract which allows users to sign up for a poll.
-abstract contract SignUpGatekeeper {
+abstract contract SignUpGatekeeper is Policy, IBasePolicy {
+  /// @notice custom errors
+  error AlreadyRegistered();
+
   /// @notice Allows to set the MACI contract
   // solhint-disable-next-line no-empty-blocks
   function setMaciInstance(address _maci) public virtual {}
```

Comment:
this becomes redundant with excubiae we have https://github.com/privacy-scaling-explorations/excubiae/blob/main/packages/contracts/contracts/policy/Policy.sol#L34C14-L34C23